graphics/fpga-graphics/verilator/main_beam.cpp

```diff
@@ -87,12 +86,25 @@ int main(int argc, char* argv[]) {
 
         // update texture once per frame at start of blanking
         if (top->sy == V_RES && top->sx == 0) {
+            // check for quit event
+            SDL_Event e;
+            if (SDL_PollEvent(&e)) {
+                if (e.type == SDL_QUIT) {
+                    break;
+                }
+            }
+
```

Comment:
Moving this here is the biggest perf win. This polls once per frame (~60 Hz) instead of polling at the pixel clock frequency (~25 MHz).